GitExtensions/Program.cs

```diff
@@ -48,6 +49,17 @@ private static void Main()
                     return UserEnvironmentInformation.GetInformation();
                 };
 
+                DiagnosticsClient.Initialize(ThisAssembly.Git.IsDirty);
+                Application.ApplicationExit += (s, e) =>
+                {
+                    DiagnosticsClient.TrackEvent("AppExit");
+                    DiagnosticsClient.OnExit();
+                };
+
+                // Report diagnostics irrespective of whether a debugger attached or not
+                AppDomain.CurrentDomain.UnhandledException += (s, e) => DiagnosticsClient.Notify((Exception)e.ExceptionObject);
+                Application.ThreadException += (s, e) => DiagnosticsClient.Notify(e.Exception);
```

Comment:
@onovotny do I need these having the following in the ApplicationInsights.config:
```
  <TelemetryModules>
    <Add Type="Microsoft.ApplicationInsights.WindowsDesktop.DeveloperModeWithDebuggerAttachedTelemetryModule, AppInsights.WindowsDesktop"/>
    <Add Type="Microsoft.ApplicationInsights.WindowsDesktop.UnhandledExceptionTelemetryModule, AppInsights.WindowsDesktop"/>
    <Add Type="Microsoft.ApplicationInsights.WindowsDesktop.UnobservedExceptionTelemetryModule, AppInsights.WindowsDesktop" />
    <!--<Add Type="Microsoft.ApplicationInsights.WindowsDesktop.FirstChanceExceptionStatisticsTelemetryModule, AppInsights.WindowsDesktop" />-->
  </TelemetryModules>
```

I peeked inside, and they looked like they are handling the same events.

Reply:
Those are handling many of the same events, it's often easier to let the initializers do it though.